apply-ParagraphStandards.py

```diff
@@ -0,0 +1,149 @@
+                                if 'MarkdownEditorOpen' in sanity_exceptions['Exceptions'][notebook_id][i['id']]:
+                                    if sanity_exceptions['Exceptions'][notebook_id][i['id']]['MarkdownEditorOpen'] == True:
+                                        i['config']['editorHide'] = False
+                                        i['config']['enabled'] = True
+                                        i['status'] = 'READY'
+                                        has_changes = True
+                else:
+                    # all other paragraph editors
+                    if tests == 'CodeBlockFix':
+                        if i['text'].startswith('%md'):
+                            print (notebook_id + ': paragraph #' + str(x) + '(' + paragraph_title + ') Editor is set to something other than MARKDOWN.')
+                        if 'editorHide' in i['config']:
+                            if i['config']['editorHide'] == True:
+                                print (notebook_id + ': paragraph #' + str(x) + '(' + paragraph_title + ') Editor IS hidden')
+                                if modify == True:
+                                    i['config']['editorHide'] = False
+                                    has_changes = True
+                        if 'enabled' in i['config']:
+                            if i['config']['enabled'] == False:
+                                print (notebook_id + ': paragraph #' + str(x) + '(' + paragraph_title + ') Run button DISABLED')
+                                if modify == True:
+                                    i['config']['enabled'] = True
+                                    has_changes = True
+                        if 'results' in i:
+                            if 'msg' in i['results']:
+                                if 'dateStarted' in i:
+                                    if 'dateFinished' in i:
+                                        print (notebook_id + ': paragraph #' + str(x) + ' Results are STORED.')
+                                        if modify == True:
+                                            i.pop('dateStarted',None)
+                                            i.pop('dateFinished',None)
+                                            has_changes = True
+                        if i['status'] != 'READY':
+                            print (notebook_id + ': paragraph #' + str(x) + ' Paragraph not marked as READY.')
+                            if modify == True:
+                                i['status'] = 'READY'
+                                has_changes = True
+                x = x + 1
+            else:
+                print (notebook_id + " paragraph #" + str(x) + " does not contain an editorMode?")
+                x = x + 1
+        if modify == True and has_changes == True:
+            print ("Saving " + notebook_id)
+            f.seek(0)
+            json.dump(data, f, indent=2, sort_keys=False)
+            f.truncate()
```

Comment:
This is quite a large block of code. Consider breaking up into some comprehensible functions along with some comments to explain what you are trying to do.

Reply:
Clearly my linter in my editor must not be working.  I'll fix this up this morning.